### PR TITLE
Check volumeattachment crd in integration tests

### PR DESCRIPTION
**Pull request by kokhang**
Fixes https://github.com/rook/rook/issues/1175

### CODE REVIEW

File: tests/framework/utils/k8s_helper.go [hunk 1]
```diff
@@ -461,6 +461,55 @@ func (k8sh *K8sHelper) IsCRDPresent(crdName string) bool {
...
+
+	cmdArgs := []string{"-n", namespace, "get", "volumeattachments", volumeAttachmentName}
+	inc := 0
+	for inc < RetryLoop {
+		_, err := k8sh.Kubectl(cmdArgs...)
+		if err == nil {
+			k8slogger.Infof("Found the VolumeAttachment resource: " + volumeAttachmentName)
+			return true
+		}
+		time.Sleep(RetryInterval * time.Second)
```
Comment (travisn) [inline tests/framework/utils/k8s_helper.go]: can you log what you're waiting for? If the test times out you want to know what was happening

File: tests/framework/utils/k8s_helper.go [hunk 2]
```diff
@@ -461,6 +461,55 @@ func (k8sh *K8sHelper) IsCRDPresent(crdName string) bool {
...
+}
+
+//IsVolumeAttachmentResourceAbsent returns true if the VolumeAttachment resource is deleted/absent within 90s else returns false
+func (k8sh *K8sHelper) IsVolumeAttachmentResourceAbsent(namespace, volumeAttachmentName string) bool {
+
+	cmdArgs := []string{"-n", namespace, "get", "volumeattachments", volumeAttachmentName}
+	inc := 0
+	for inc < RetryLoop {
+		_, err := k8sh.Kubectl(cmdArgs...)
+		if err != nil {
```
Comment (jbw976) [inline tests/framework/utils/k8s_helper.go]: do we need to check the error further here to be sure that we are being told the volumeattachment doesn't exist versus some other error? as in `errors.IsNotFound(err)`

Comment (travisn) [inline tests/framework/utils/k8s_helper.go]: you'll want to check for `errors.IsNotFound(err)`

File: tests/framework/utils/k8s_helper.go [hunk 3]
```diff
@@ -461,6 +461,55 @@ func (k8sh *K8sHelper) IsCRDPresent(crdName string) bool {
...
+		}
+		time.Sleep(RetryInterval * time.Second)
+		inc++
+	}
+
+	return false
+}
+
+//IsVolumeAttachmentResourceAbsent returns true if the VolumeAttachment resource is deleted/absent within 90s else returns false
+func (k8sh *K8sHelper) IsVolumeAttachmentResourceAbsent(namespace, volumeAttachmentName string) bool {
```
Comment (travisn) [inline tests/framework/utils/k8s_helper.go]: these two test methods are exactly the same, except how to check for the error returned. can you refactor? for example, an argument could indicate if you're waiting for it to exist or be deleted, then the error checking is just a little different.

File: tests/framework/utils/k8s_helper.go (hunk 2 above)
Comment (kokhang) [reply tests/framework/utils/k8s_helper.go]: i thought about it but we shouldnt get `any` errors. If we get an error, whether it is a NotFound or some other error, the test should fail.

File: tests/framework/utils/k8s_helper.go (hunk 3 above)
Comment (kokhang) [reply tests/framework/utils/k8s_helper.go]: But that different is what makes the methods to the different :P

File: tests/framework/utils/k8s_helper.go (hunk 2 above)
Comment (kokhang) [reply tests/framework/utils/k8s_helper.go]: sounds good

Comment (kokhang) [reply tests/framework/utils/k8s_helper.go]: oh yes.. i thought that was for a different function. Will do. Thanks

**Comment by kokhang** [comment]
ahh i cant use the errors.IsNotFound(err) since it is executing kubectl and not doing it via client-go bindings.

**Comment by travisn** [comment]
@kokhang actually can you use the k8s api directly instead of kubectl? We've been trying to get away from kubectl in the tests where possible

**Comment by kokhang** [comment]
@travisn I incorporated all your feedback.. PTAL

**Comment by kokhang** [comment]
By the way, I'm using the restclient as supposed to the go bindings to get the Volume attachment crd because it works both for Crd and TPR.

**Pull request merged**